pkg/runtime/runtime.go

```diff
@@ -758,6 +780,45 @@ func (a *DaprRuntime) initPubSub() error {
 	return nil
 }
 
+// Publish is an adapter method for the runtime to pre-validate publish requests
+// And then forward them to the Pub/Sub component.
+// This method is used by the HTTP and gRPC APIs.
+func (a *DaprRuntime) Publish(req *pubsub.PublishRequest) error {
+	if allowed := a.isPubSubOperationAllowed(req.Topic, a.allowedPublishings); !allowed {
+		return fmt.Errorf("topic %s is not allowed for app id %s", req.Topic, a.runtimeConfig.ID)
+	}
+	return a.pubSub.Publish(req)
+}
+
+func (a *DaprRuntime) isPubSubOperationAllowed(topic string, topicsList []string) bool {
+	allowed := false
+
+	// first check if allowedTopics contain it
+	if len(a.allowedTopics) > 0 {
+		for _, t := range a.allowedTopics {
+			if t == topic {
```

Comment:
nit: assumption is casing making different topic here, right ?

Reply:
correct